<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AIAgent</class>
 <widget class="QDialog" name="AIAgent">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="focusPolicy">
   <enum>Qt::FocusPolicy::WheelFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Edge AI Agent Application</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::WeatherClear"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="spacing">
    <number>9</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="focusPolicy">
      <enum>Qt::FocusPolicy::WheelFocus</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="WndConnect">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <attribute name="icon">
       <iconset theme="network-offline"/>
      </attribute>
      <attribute name="title">
       <string>Router Connection:</string>
      </attribute>
      <attribute name="toolTip">
       <string>Router Connection</string>
      </attribute>
      <attribute name="whatsThis">
       <string>Multi-Target Router Connection</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Multi-Target Router Connection:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <property name="horizontalSpacing">
           <number>12</number>
          </property>
          <property name="verticalSpacing">
           <number>16</number>
          </property>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Location of Models:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="TxtTextLen"/>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Max. Text Length:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QListWidget" name="ListModels">
            <property name="editTriggers">
             <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
            </property>
            <property name="showDropIndicator" stdset="0">
             <bool>false</bool>
            </property>
            <property name="spacing">
             <number>1</number>
            </property>
            <property name="selectionRectVisible">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="RouterPort">
            <property name="toolTip">
             <string>MTRouter Port Number</string>
            </property>
            <property name="maxLength">
             <number>4</number>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QWidget" name="widget" native="true">
            <widget class="QPushButton" name="BtnActivate">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>121</width>
               <height>29</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Activate</string>
             </property>
            </widget>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>40</y>
               <width>121</width>
               <height>141</height>
              </rect>
             </property>
             <property name="title">
              <string>Reply quality:</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <property name="spacing">
               <number>3</number>
              </property>
              <property name="leftMargin">
               <number>7</number>
              </property>
              <property name="topMargin">
               <number>7</number>
              </property>
              <property name="rightMargin">
               <number>7</number>
              </property>
              <property name="bottomMargin">
               <number>9</number>
              </property>
              <item>
               <widget class="QRadioButton" name="BtnBest">
                <property name="text">
                 <string>The Best</string>
                </property>
                <property name="iconSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="BtnBetter">
                <property name="text">
                 <string>Better</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="BtnAverage">
                <property name="text">
                 <string>Average</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="BtnFast">
                <property name="text">
                 <string>Fast</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="BtnFastest">
                <property name="text">
                 <string>The Fastest</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="RouterAddress">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>MTRouter IP Address or Host Name</string>
            </property>
            <property name="maxLength">
             <number>512</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>AI LLaMA Models:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Router IP Address:</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Max. Tokens:</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="6" column="1">
           <widget class="QLineEdit" name="TxtTokens"/>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="TxtModelDir">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Router Port Number:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="BtnConnect">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>30</height>
             </size>
            </property>
            <property name="sizeIncrement">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="baseSize">
             <size>
              <width>120</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>&amp;Connect</string>
            </property>
            <property name="icon">
             <iconset theme="network-wireless"/>
            </property>
            <property name="iconSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
            <property name="shortcut">
             <string>Alt+C</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="default">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QPushButton" name="BtnBrowse">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="WndChat">
      <attribute name="icon">
       <iconset theme="user-available"/>
      </attribute>
      <attribute name="title">
       <string>AI Agent Chat:</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Chat History:</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>2</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>2</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <widget class="QTableView" name="TableHistory">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
            </property>
            <property name="tabKeyNavigation">
             <bool>false</bool>
            </property>
            <property name="showDropIndicator" stdset="0">
             <bool>false</bool>
            </property>
            <property name="dragDropOverwriteMode">
             <bool>false</bool>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
            </property>
            <property name="iconSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
            <attribute name="horizontalHeaderCascadingSectionResizes">
             <bool>true</bool>
            </attribute>
            <attribute name="horizontalHeaderMinimumSectionSize">
             <number>50</number>
            </attribute>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>150</number>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderHighlightSections">
             <bool>false</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="TxtDisplay">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>90</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QWidget" name="widget_2" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>18</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>312</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>3</number>
        </property>
        <property name="leftMargin">
         <number>7</number>
        </property>
        <property name="topMargin">
         <number>7</number>
        </property>
        <property name="rightMargin">
         <number>7</number>
        </property>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Type:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="TxtAgentType">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Queue Size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="TxtQueueSize">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Active Model:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="TxtActiveModel">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="BtnClose">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>120</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>120</width>
         <height>30</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>120</width>
         <height>30</height>
        </size>
       </property>
       <property name="layoutDirection">
        <enum>Qt::LayoutDirection::LeftToRight</enum>
       </property>
       <property name="text">
        <string>Close</string>
       </property>
       <property name="icon">
        <iconset theme="application-exit"/>
       </property>
       <property name="iconSize">
        <size>
         <width>12</width>
         <height>12</height>
        </size>
       </property>
       <property name="shortcut">
        <string>Alt+X</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>RouterAddress</tabstop>
  <tabstop>RouterPort</tabstop>
  <tabstop>BtnConnect</tabstop>
  <tabstop>BtnClose</tabstop>
  <tabstop>TableHistory</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
